event_listener/dfusion_db/event_pusher.py

```diff
@@ -69,11 +78,34 @@ def update_accounts(event: Dict[str, Union[int, str, str, int]]):
         db.accounts.insert_one(new_account_record)
 
     elif transition_type == 1:  # Withdraw
```

Comment:
I wonder if this code could be made a bit cleaner and shared with deposits (it looks 50% similar). 

Nit: can transition_type be an enum to make the documentation explicit?

Reply:
Yes I like the Enum idea, but I still think deposit and withdraw logic are different and should be kept separate. This is due to the fact that some withdraw requests are not processed (i.e. due to insufficient balances mentioned in another place).

Reply:
I think this logic should be shared, since it's identical:

```
a_id = deposit['accountId']
t_id = deposit['tokenId']
amount = deposit['amount']
index = num_tokens * (a_id - 1) + (t_id - 1)
```